cdap-app-fabric/src/main/java/co/cask/cdap/internal/app/runtime/spark/SparkUtils.java

```diff
@@ -155,6 +188,158 @@ public static URLClassLoader createSparkFrameworkClassLoader(ClassLoader parentC
     return new URLClassLoader(urls, parentClassLoader);
   }
 
+  /**
+   * Creates a Zip file which contains two files inside, a "yarn-site.xml", which is serialization of the given
+   * {@link YarnConfiguration}, and a "__spark_conf__.properties", which contains the serialization of the given
+   * spark properties
+   *
+   * @param yarnConf the YARN configuration to serialize
+   * @param sparkProperties the Spark configuration to serialize
+   * @param confZip file the target file to write to
+   * @return the given confZip file
+   * @throws IOException if failed to write to the zip file
+   */
+  public static File createSparkConfZip(YarnConfiguration yarnConf, Properties sparkProperties,
+                                        File confZip) throws IOException {
+    try (ZipOutputStream zipOutput = new ZipOutputStream(new FileOutputStream(confZip))) {
+      zipOutput.putNextEntry(new ZipEntry("yarn-site.xml"));
+      yarnConf.writeXml(zipOutput);
+
+      zipOutput.putNextEntry(new ZipEntry(SPARK_CONF_FILE));
+      sparkProperties.store(zipOutput, "Spark configuration.");
+    }
+    return confZip;
+  }
+
+  /**
+   * Returns the Spark Jar location with the Spark Yarn Client rewritten. It is for workaround the bug in
+   * CDAP-5019 (SPARK-13441).
```

Comment:
Your method says it returns "A rewritten Spark assembly JAR", but your Javadoc says "Spark Jar Location". Which is it?

If the location:
```
Returns the location of the Spark Jar with the rewritten Spark Yarn Client. 
It is a workaround for the bug in CDAP-5019 (SPARK-13441).
```
If the Jar:
```
Returns the Spark Jar with the rewritten Spark Yarn Client. It is a workaround for the bug in
CDAP-5019 (SPARK-13441).